flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/TableSchema.java

```diff
@@ -343,13 +404,36 @@ public Builder field(String name, TypeInformation<?> typeInfo) {
 			return field(name, fromLegacyInfoToDataType(typeInfo));
 		}
 
+		/**
+		 * Specifies the previously defined field as an event-time attribute and specifies the watermark strategy.
+		 *
+		 * @param rowtimeAttribute the field name as a rowtime attribute, can be a nested field using dot separator.
+		 * @param watermarkExpressionString the string representation of watermark generation expression,
+		 *                                  e.g. "ts - INTERVAL '5' SECOND". The string is a qualified SQL expression
+		 *                                  string (UDFs are expanded) but will not be validated by {@link TableSchema}.
+		 * @param watermarkExprOutputType the data type of the computation result of watermark generation expression.
+		 *                                Whether the data type equals to the output type of expression will also
+		 *                                not be validated by {@link TableSchema}.
+		 */
+		public Builder watermark(String rowtimeAttribute, String watermarkExpressionString, DataType watermarkExprOutputType) {
+			Preconditions.checkNotNull(rowtimeAttribute);
```

Comment:
`watermarkExpressionString` -> `watermarkExpr`, the type already indicates that it is a String;
`watermarkExprOutputType` -> `watermarkExprType` obviously the type of a expression is it's output type.

Reply:
We don't run out of characters. A better field name describes everything. If we use `watermarkExpr`, some users may guess it is an `Expression` but is a `String`.